pkg/cmd/cli/cmd/create/deploymentconfig.go

```diff
@@ -0,0 +1,141 @@
+package create
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+const (
+	DeploymentConfigRecommendedName = "deploymentconfig"
+
+	deploymentConfigLong = `
+Create a deployment config that uses a given image.`
+
+	deploymentConfigExample = `  # Create an nginx deployment config named my-nginx
+  $ %[1]s my-nginx --image=nginx`
+)
+
+type CreateDeploymentConfigOptions struct {
+	DC     *deployapi.DeploymentConfig
+	Client client.DeploymentConfigsNamespacer
+
+	Mapper       meta.RESTMapper
+	OutputFormat string
+	Out          io.Writer
+	Printer      ObjectPrinter
+}
+
+// NewCmdCreateServiceAccount is a macro command to create a new service account
+func NewCmdCreateDeploymentConfig(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &CreateDeploymentConfigOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:     name + " NAME --image=IMAGE -- [COMMAND] [args...]",
+		Short:   "Create deployment config with default options that uses a given image.",
+		Long:    policyBindingLong,
+		Example: fmt.Sprintf(policyBindingExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, f, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Aliases: []string{"dc"},
+	}
+
+	cmd.Flags().String("image", "", "The image for the container to run.")
+
+	cmdutil.AddOutputFlagsForMutation(cmd)
+	return cmd
+}
+
+func (o *CreateDeploymentConfigOptions) Complete(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("at least one argument is required")
+	}
+	labels := map[string]string{"from-dc": args[0]}
+
+	o.DC = &deployapi.DeploymentConfig{
+		ObjectMeta: kapi.ObjectMeta{Name: args[0]},
+		Spec: deployapi.DeploymentConfigSpec{
+			Selector: labels,
```

Comment:
I don't think we should add a pre-configured selector. Have a second flag for this and require it.